### DISCUSSION TITLE
Puma::ConnectionError: Socket timeout writing data

### DISCUSSION THREAD

**Discussion by qelphybox** [puma/puma#3445] [Q&A]
After the last deploy, the application intermittently triggers a `Puma::ConnectionError` on random requests:
```
Puma::ConnectionError: Socket timeout writing data
puma (5.6.7) lib/puma/request.rb:219:in `rescue in fast_write'
puma (5.6.7) lib/puma/request.rb:210:in `fast_write'
puma (5.6.7) lib/puma/request.rb:152:in `block in handle_request'
rack (2.2.9) lib/rack/body_proxy.rb:41:in `each'
rack (2.2.9) lib/rack/body_proxy.rb:41:in `method_missing'
```
The error does not seem to impact user experience, as the server continues to respond to requests. However, it fills the logs with this error message.  What could be a reason? or  how can I find out the cause? please help:)

**Puma config:**

```ruby
threads_count = ENV.fetch('RAILS_MAX_THREADS', 5)
threads threads_count, threads_count

port ENV.fetch('PORT', 3000)
environment ENV.fetch('RAILS_ENV', 'development')
workers ENV.fetch('WEB_CONCURRENCY', 1)

plugin :tmp_restart
```

**Desktop (please complete the following information):**
- OS: Ubuntu 20.04
- Puma Version: 5.6.7
- Docker Version: 24.0.6

**Comment by dentarg** [comment]
Maybe look into what changed with the last deploy?

Puma issues aren't really for debugging your deployment/application. I converted this to a discussion.
